crates/editor/src/editor.rs

```diff
@@ -11678,6 +11678,41 @@ impl Editor {
                         let is_within_comment_override = buffer
                             .language_scope_at(indent_end)
                             .is_some_and(|scope| scope.override_name() == Some("comment"));
+
+                        if let Some((_, range)) = buffer.syntax_ancestor(indent_end..indent_end) {
+                            let text_range = match range {
+                                MultiOrSingleBufferOffsetRange::Multi(r) => Some(r),
+                                MultiOrSingleBufferOffsetRange::Single(_) => None,
+                            };
+                            if let Some(range) = text_range {
```

Comment:
Maybe we don't even need to get buffer line, simply get text iter from this range, and check prefix directly. Because this range will always start with comment prefix (i.e. no whitespace in between).